frontend/src/components/worksheets/Worksheet/Worksheet.js

```diff
@@ -1507,11 +1512,31 @@ class Worksheet extends React.Component {
                             this.setFocus(focus, this.state.subFocusIndex);
                         }
                     }
-                    this.setState({
-                        updating: false,
-                        version: this.state.version + 1,
-                        numOfBundles: numOfBundles,
-                    });
+                    this.setState(
+                        {
+                            updating: false,
+                            version: this.state.version + 1,
+                            numOfBundles: numOfBundles,
+                        },
+                        () => {
+                            if (addImage) {
+                                const subFocusIndex = this.state.subFocusIndex
```

Comment:
Let's simplify this to
```suggestion
                                const subFocusIndex = this.state.subFocusIndex || 0;
```